pkgs/tools/networking/opensm/default.nix

```diff
@@ -16,6 +17,8 @@ stdenv.mkDerivation rec {
 
   preConfigure = "bash ./autogen.sh";
```

Comment:
Not sure if this might be a source of problems but this could be simpler:
```rust
preConfigure = "./autogen.sh"
```

Let it be run in the current build env (sandboxed) with the current shell.

Reply:
It should not cause problems but I agree that it is unnecessary.

Reply:
Yes it's done in plenty of packages.
At the moment [the source](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/patch-shebangs.sh) states:

> This setup hook [...] rewrite(s) all script interpreter file names to paths found in $PATH

Seems safe to use it that way.

Reply:
@c0bw3b I am still curious what would actually happen in a cross environment. `nativeBuildInputs` and `buildInputs` determine what should be run time and what should be pure build time dependency. These parts should be clear then. However, `bash` is part of the build env, such that the `fixupPhase` would write a `bash` into scripts that belongs to the build env instead the target env? 
I am sure nix has some magic going on that fixes it correctly but it is not obvious to me.

Reply:
The build env is sandboxed but still using a bash from the nix store, I'm guessing?

I'm not well-versed in cross-compilation matters but I see that one commit on `patch-shebangs.sh` to address cross-compile was then reverted.
In any way the nix function `patchShebangs` will "mask" the complexity and decide what to do when cross-compiling or not.

Reply:
Aha

> # This setup hook causes the fixup phase to rewrite all script
> [...]
> `newPath="$(command -v "$(basename "$oldPath")" || true)"

So, as is, it should be fine to use for patching build scripts.

When executed during the fixup phase (in the target env), it will patch to use the target env, but when executed during preConfigure (in the build env), it will actually patch to use the build env.

Reply:
There is the nix magic! Thanks for digging into it.